packages/odyssey-react-mui/src/ui-shell/UiShell.tsx

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  memo,
+  useEffect,
+  useState,
+  type ReactNode,
+  type SetStateAction,
+} from "react";
+import { ErrorBoundary } from "react-error-boundary";
+
+import { SideNav, type SideNavProps } from "../labs/SideNav";
+import { TopNav, type TopNavProps } from "../labs/TopNav";
+import { OdysseyProvider } from "../OdysseyProvider";
+import { ShadowDomElements } from "../web-component/shadow-dom";
+
+const appContainerStyles = {
+  flexGrow: "1",
+};
+
+const uiShellContainerStyles = {
+  display: "flex",
+};
+
+export type UiShellComponentProps = {
+  sideNavProps: Omit<SideNavProps, "logo" | "footerComponent">;
+  topNavProps: Omit<
+    TopNavProps,
+    "AdditionalNavItemComponent" | "SearchFieldComponent"
+  >;
+};
+
+export const defaultComponentProps: UiShellComponentProps = {
+  sideNavProps: {
+    navHeaderText: "",
+    sideNavItems: [],
+  },
+  topNavProps: {
+    topNavLinkItems: [],
+  },
+};
+
+export type UiShellProps = {
```

Comment:
Think we could/should add some doc comments here to help clarify what these are?

Also, I see `appRootElement` is used below but is not listed here. That's the principal item that I'm curious about doc-ing: I see in `renderUiShell` that you pass in an empty `<slot />` element for `appComponent` and trying to understand what's going on there.

Reply:
Great point! I was missing docs on the props. That would help a ton provided you have those types available in your app.

---

I'm confused to what you're asking.

`appRootElement` isn't used by `UiShell`, but there's `appComponent`; a `<slot />`. I render that slot in the correct spot, then in `renderUiShell`, it appends `appRootElement` to the React Web Component which puts it in that slot passed to `appComponent` in `UiShell`.

Reply:
What I'm asking: Here in defining `UiShellProps` there is no `appRootElement`, but there is one defined on L72 as part of the props object. I'm trying to understand if that's a prop that should be listed here in `UiShellProps` (perhaps with doc as well)

Reply:
Docs that may help clarify what you're asking: f2a373f3391a6bfbfece4d6f6eced735b7a2508e.